lib/bemxjst/tree.js

```diff
@@ -195,66 +191,20 @@ AddMatch.prototype.mixWrapBody = function mixWrapBody(body) {
   };
 };
 
-AddMatch.prototype.attrsWrapBody = function attrsWrapBody(body) {
-  var apply = this.refs.apply;
-
-  if (typeof body !== 'function') {
-    return function inlineAddAttrsAdaptor() {
-      return utils.extend(apply('attrs') || {}, body);
-    };
-  }
-
-  return function addAttrsAdaptor() {
-    return utils.extend(apply('attrs') || {}, body.call(this, this, this.ctx));
-  };
-};
-
-AddMatch.prototype.jsWrapBody = function jsWrapBody(body) {
-  var apply = this.refs.apply;
+[ 'attrs', 'js', 'mods', 'elemMods' ].forEach(function(method) {
+  AddMatch.prototype[ method + 'WrapBody'] = function(body) {
+    var apply = this.refs.apply;
 
-  if (typeof body !== 'function') {
-    return function inlineAddJsAdaptor() {
-      return utils.extend(apply('js') || {}, body);
-    };
-  }
-
-  return function addJsAdaptor() {
-    return utils.extend(apply('js') || {}, body.call(this, this, this.ctx));
-  };
-};
-
-AddMatch.prototype.modsWrapBody = function modsWrapBody(body) {
-  var apply = this.refs.apply;
-
-  if (typeof body !== 'function') {
-    return function inlineAddModsAdaptor() {
-      this.mods = utils.extend(apply('mods'), body);
-      return this.mods;
-    };
-  }
-
-  return function addModsAdaptor() {
-    this.mods = utils.extend(apply('mods'), body.call(this, this, this.ctx));
-    return this.mods;
-  };
-};
-
-AddMatch.prototype.elemModsWrapBody = function elemModsWrapBody(body) {
-  var apply = this.refs.apply;
-
-  if (typeof body !== 'function') {
-    return function inlineAddElemModsAdaptor() {
-      this.elemMods = utils.extend(apply('elemMods'), body);
-      return this.elemMods;
-    };
-  }
-
-  return function addElemModsAdaptor() {
-    this.elemMods = utils.extend(apply('elemMods'),
-                             body.call(this, this, this.ctx));
-    return this.elemMods;
+    return typeof body !== 'function' ?
```

Comment:
It’s not the same.

`function with ternary` VS `generator for different functions`.